compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -660,9 +660,18 @@ impl<'a> Parser<'a> {
     fn recover_not_expr(&mut self, lo: Span) -> PResult<'a, (Span, ExprKind)> {
         // Emit the error...
         let negated_token = self.look_ahead(1, |t| t.clone());
+        let negtated_msg = if negated_token.is_numeric_lit() {
```

Comment:
I don't think we should be passing this in as an english phrase -- either use a [fluent selector](https://projectfluent.org/fluent/guide/selectors.html) or split this into several different suggestion messages.

Reply:
Fixed, thanks @compiler-errors.